linux57-tkg/install.sh

```diff
@@ -132,74 +167,118 @@ if [ "$1" = "install" ]; then
+      fi
+      
+      # Move rpm files to RPMS folder inside the linux-tkg folder
+      mv "$_where"/*.deb "$_where"/DEBS/
+
       read -p "Do you want to install the new Kernel ? y/[n]: " _install
-      if [[ $_install =~ [yY] ]] || [[ $_install =~ [yY] ]] || [ $_install = "yes" ] || [ $_install = "Yes" ]; then
+      if [[ $_install =~ [yY] ]] || [ $_install = "yes" ] || [ $_install = "Yes" ]; then
         cd "$_where"
         _kernelname=$_basekernel.$_kernel_subver-$_kernel_flavor
-        _headers_deb=linux-headers-${_kernelname}*.deb
-        _image_deb=linux-image-${_kernelname}_*.deb
+        _headers_deb="linux-headers-${_kernelname}*.deb"
+        _image_deb="linux-image-${_kernelname}_*.deb"
+        _kernel_devel_deb="linux-libc-dev_${_kernelname}*.rpm"
         
-        sudo dpkg -i $_headers_deb $_image_deb
+        sudo dpkg -i DEBS/$_headers_deb DEBS/$_image_deb DEBS/$_kernel_devel_deb
+      fi
+    fi
 
-        # Add to the list of installed kernels, used for uninstall
-        if ! { [ -f installed-kernels ] && grep -Fxq "$_kernelname" installed-kernels; }; then
-          echo $_kernelname >> installed-kernels 
-        fi   
+  elif [[ "$_distro" = "Fedora" ||  "$_distro" = "Suse" ]]; then
+
+    # Replace dashes with underscores, it seems that it's being done by binrpm-pkg
+    # Se we can actually refer properly to the rpm files.
+    _kernel_flavor=${_kernel_flavor//-/_}
+
+    if make -j ${_thread_num} rpm-pkg EXTRAVERSION="_${_kernel_flavor}"; then
+      msg2 "Building successfully finished!"
+
+      cd "$_where"
+
+      # Create rpms folder if it doensn't exist
+      if ! [ -d RPMS ]; then
+        mkdir RPMS
+      fi
+      
+      # Move rpm files to RPMS folder inside the linux-tkg folder
+      mv ~/rpmbuild/RPMS/x86_64/* "$_where"/RPMS/
+
+      #Clean up the original folder, unneeded and takes a lot of space
+      rm -rf ~/rpmbuild/
+
+      read -p "Do you want to install the new Kernel ? y/[n]: " _install
+      if [ "$_install" = "y" ] || [ "$_install" = "Y" ] || [ "$_install" = "yes" ] || [ "$_install" = "Yes" ]; then
+        
+        _kernelname=$_basekernel.${_kernel_subver}_$_kernel_flavor
+        _headers_rpm="kernel-headers-${_kernelname}*.rpm"
+        _kernel_rpm="kernel-${_kernelname}*.rpm"
+        _kernel_devel_rpm="kernel-devel-${_kernelname}*.rpm"
+        
+        if [ "$_distro" = "Fedora" ]; then
+          sudo dnf install RPMS/$_headers_rpm RPMS/$_kernel_rpm RPMS/$_kernel_devel_rpm
+        elif [ "$_distro" = "Suse" ]; then
+          msg2 "Some files from 'linux-glibc-devel' will be replaced by files from the custom kernel-hearders package"
+          msg2 "To revert back to the original kernel headers do 'sudo zypper install -f linux-glibc-devel'" 
+          sudo zypper install --replacefiles --allow-unsigned-rpm RPMS/$_headers_rpm RPMS/$_kernel_rpm RPMS/$_kernel_devel_rpm
+        fi
+        
+        msg2 "Install successful" 
       fi
     fi
   fi
+
```

Comment:
Unneeded newline (sorry :3).

Reply:
It was done in purpose actually! I like to have a free space after the `if` condition and before the `fi` closing keyword. But I agree here in particular it kills the nice nesting.